web-ng/src/app/components/text-field/text-field.component.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Component,
+  OnInit,
+  Input,
+  Output,
+  EventEmitter,
+  OnChanges,
+} from '@angular/core';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import { FieldType } from '../../shared/models/form/field.model';
+
+export interface FieldTypeOptionModel {
+  icon: string;
+  label: string;
+  type: FieldType;
+}
+
+@Component({
+  selector: 'app-text-field',
+  templateUrl: './text-field.component.html',
+  styleUrls: ['./text-field.component.css'],
+})
+export class TextFieldComponent implements OnInit, OnChanges {
```

Comment:
Are we sure breaking these UI components down along the lines of field type is the best approach here? The label, type, and required toggle are the same for all fields, so it would seem fitting if there was one "FieldEditorComponent", which in turn contains other [potentially hidden] subcomponents (e.g., OptionList, OptionSelector, etc) depending on the field type. Wdyt?